### DISCUSSION TITLE
Multiple executeDataQuery calls in parallel with YDB Java sdk.

### DISCUSSION THREAD

**Discussion by oleg68** [ydb-platform/ydb#67] [Q&A] [answered]
Because Session.executeDataQuery returns CompletableFuture, seems it is intended to call executeDataQuery asynchronously in parallel and then to combine their results.

But when I do so, I receive an exception ``com.yandex.ydb.core.UnexpectedResultException: Execute select, code: SESSION_BUSY, issues: [Pending previous query completion (S_ERROR)]``

Is it possible to have several outstanding data query operations within one transaction?

**Comment by spuchin** [comment, marked as answer]
No, currently you cannot execute multiple query operations in parallel on single session (transaction). Still you can get multiple sessions and execute queries in parallel in separate sessions.
